### PR TITLE
Update devkit to v0.3

### PR DESCRIPTION
**Pull request by patk-motional**
Version bump to nuPlan devkit v0.3

Feature list:

- Metrics run time improvements @shakiba-motional @Noushin.Mehdipour
- Refactored database models to improve test coverage @Rachel.Koh @Clarence.Chye
- Add lane boundary API to the maps @christopher-motional 
- Model deployment pipeline @Mitchell.Spryn
- Reduce RAM usage @Mitchell.Spryn
- Devkit setup fixes @gianmarco-motional 
- nuBoard improvements @kokseang-motional 
- Improve simulation runtime @michael-noronha-motional 
- Kinematic bicycle model improvements @shakiba-motional 
- Enable LR schedulers @Hiok.Hian.Ong

### CODE REVIEW

**Comment by kokseang-motional** [comment]
Update `nuplan_scenario_visualization.ipynb` to load SimulationLog

**Review by gianmarco-motional — APPROVED** [review]
Approved for my side!

File: nuplan/planning/training/experiments/caching.py [hunk 1]
```diff
@@ -1,49 +1,126 @@
...
-from typing import List
+import os
+import uuid
+from typing import Dict, List, Union, cast
 
-from omegaconf import DictConfig
+from omegaconf import DictConfig, OmegaConf
 
+from nuplan_devkit.nuplan.common.utils.s3_utils import check_s3_path_exists
+from nuplan_devkit.nuplan.database.nuplan_db.nuplandb_wrapper import get_db_filenames_from_load_path
```
Comment (juraj-motional) [inline nuplan/planning/training/experiments/caching.py:10]: why is here nuplan_devkit?

**Review by christopher-motional — APPROVED** [review]
NuPlan planner tutorial works as expected.

**Comment by patk-motional** [comment]
Close due to CI being down. Will update with a newer version

**Pull request closed without merging**